crates/storage/libmdbx-rs/src/error.rs

```diff
@@ -224,8 +226,8 @@ pub(crate) fn mdbx_result_with_tx_kind<K: TransactionKind>(
     txn_manager: &TxnManager,
 ) -> Result<bool> {
     if K::IS_READ_ONLY &&
-        err_code == ffi::MDBX_EBADSIGN &&
-        txn_manager.remove_aborted_read_transaction(txn).is_some()
+        txn_manager.remove_aborted_read_transaction(txn).is_some() &&
+        err_code == ffi::MDBX_EBADSIGN
```

Comment:
can you please explain this change?

Reply:
it allows a txn ptr to be reused if mdbx deems it fit to reuse, i.e. error code is 0

Reply:
I think otherwise we create dangling references right? if `err_code == 0` but we still return `Error::ReadTransactionAborted`

Reply:
if `err_code == 0`, but the txn stays in aborted txns, then aborting it will lead to `Error::ReadTransactionAborted`

Reply:
if the error code is not `MDBX_EBADSIGN`, why do we want to remove it from the list of aborted transactions?

Reply:
no wait got that mixed up.

txn is aborted because time out, makes mdbx say "hey this ptr can be assigned"
open "new" txn and mdbx uses that same ptr, so err code is `0`, so that check fails already at `err_code == ffi::MDBX_EBADSIGN`
txn stays in aborted list
user tries to abort the "new" txn, gets `Error::TransactionAborted`, very confused user

Reply:
makes more sense to change order of that guard I think, then to remove from the aborted list when user tries to abort by drop or by `TxnManagerMessage`

Reply:
> user tries to abort the "new" txn, gets Error::TransactionAborted, very confused user

why would that happen? If it tries to abort a new transaction, it will go through the `mdbx_txn_abort` just fine and return a code `0`, I think?

Reply:
mode this to its own scope https://github.com/paradigmxyz/reth/pull/6844